src/plugins/philsPluginLibrary/store/store.ts

```diff
@@ -32,23 +32,86 @@ export interface PluginStore<Z extends PluginSettings> {
     set: PluginSet<Z>;
 }
 
+function createObjectProxy<T extends object>(obj1: T, onUpdate: (updatedObject: T) => void): T {
+    const handler: ProxyHandler<T> = {
+        set(target, property, value, receiver) {
+            const success = Reflect.set(target, property, value, receiver);
+            const nestedObj = target[property];
+
+            if (typeof nestedObj == 'object') {
+                target[property] = createObjectProxy(nestedObj, () => { onUpdate(obj1); }); //On update will call itself until the top level object
+            }
+
+            onUpdate(obj1); //This will recursively call on nested objects
+            return success;
+        }
+    };
+
+    return new Proxy(obj1, handler);
+}
+
+
+const startupStates = {};
+const settingStorage = new Map();
 export function createPluginStore<Z extends PluginSettings = {}>(pluginName: string, storeName: string, f: PluginInitializer<Z>): PluginStore<Z> {
     if (!Settings.plugins[pluginName])
         throw new Error("The specified plugin does not exist");
 
     if (!Settings.plugins[pluginName].stores)
         Settings.plugins[pluginName].stores = {};
 
-    const get: PluginGet<Z> = () => Settings.plugins[pluginName].stores[storeName] as Z;
+    if (!Settings.plugins[pluginName].stores[storeName]) //Just incase the store doesn't exist we create it here (otherwise we crash)
+        Settings.plugins[pluginName].stores[storeName] = {};
+
+    const get: PluginGet<Z> = () => {
+        const storeSettings = settingStorage.get(storeName);
+        if (!startupStates[storeName]) { //We do this so that we can load all the saved data without the proxy attempting to overwrite it
+            const startupInfo = Settings.plugins[pluginName].stores[storeName];
+
+            storeSettings.simpleMode = startupInfo.simpleMode;
+            storeSettings.profiles = startupInfo.profiles || [];
+            storeSettings.currentProfile = startupInfo.currentProfile || { name: '' };
+
+            startupStates[storeName] = true;
+        }
+
+        return storeSettings;
+    };
+
     const set: PluginSet<Z> = (s: ((settings: Z) => Z | undefined) | Z) =>
-        Settings.plugins[pluginName].stores[storeName] = (typeof s === "function" ? s(get()) : s) || get();
```

Comment:
Why was the `Settings.plugins[pluginName].stores[storeName] =` statement removed? Isn't it still necessary?

Reply:
Hello, thank you for your response. I'll make it use the JSON method and update the pull request.
I removed the set because it would be passing function into the object and would cause the plugin to crash. The same behavior is in the updateCallback except it's no longer passing functions to it.